onnxruntime/core/providers/dml/DmlExecutionProvider/src/DmlCommandRecorder.cpp

```diff
@@ -15,11 +15,17 @@ DmlCommandRecorder::DmlCommandRecorder(
     : m_queue(std::move(commandQueue)),
       m_d3dDevice(d3dDevice),
       m_dmlDevice(dmlDevice),
-      m_descriptorPool(d3dDevice, 2048),
-      m_commandAllocatorRing(d3dDevice, m_queue->GetType(), m_queue->GetCurrentCompletionEvent())
+      m_descriptorPool(d3dDevice, 2048)
 {
     ORT_THROW_IF_FAILED(dmlDevice->CreateOperatorInitializer(0, nullptr, IID_PPV_ARGS(&m_initializer)));
     ORT_THROW_IF_FAILED(dmlDevice->CreateCommandRecorder(IID_PPV_ARGS(&m_recorder)));
+
+    m_threadPool = std::make_unique<onnxruntime::concurrency::ThreadPool>(
```

Comment:
I *think* we want to re-use other thread pool instances and in doing so respect API options for thread count.  May be a good idea to get input from ORT devs on this.